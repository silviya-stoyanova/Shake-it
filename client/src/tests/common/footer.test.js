import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { mount } from 'enzyme'
import Footer from '../../components/common/footer'

describe('tests for the component Footer', () => {
    it('should render properly', () => {
        const wrapper = mount(
            <BrowserRouter>
                <Footer />
            </BrowserRouter>
        )
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should have a div element with className "left-al"', () => {
        const wrapper = mount(
            <BrowserRouter>
                <Footer />
            </BrowserRouter>
        )
        expect(wrapper.find("div[className='left-al']").length).toEqual(1)
    })

    it('should have a div element with className "right-al"', () => {
        const wrapper = mount(
            <BrowserRouter>
                <Footer />
            </BrowserRouter>
        )
        expect(wrapper.find("div[className='right-al']").length).toEqual(1)
    })

    it('should have 4 span elements with proper text', () => {
        const spanElsText = ['Terms', 'Data use policy', 'Make contact with us', 'Shake it, Bulgaria']

        const wrapper = mount(
            <BrowserRouter>
                <Footer />
            </BrowserRouter>
        )

        wrapper.find('span').map((span, index) => {
            expect(span.text()).toMatch(new RegExp(`${spanElsText[index]}`, 'i'))
        })
    })
})