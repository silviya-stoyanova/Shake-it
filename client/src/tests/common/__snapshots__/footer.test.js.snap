// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for the component Footer should render properly 1`] = `"<footer><div class=\\"left-al\\"><span><a href=\\"/terms\\">Terms</a></span><span><a href=\\"/data-use-policy\\">Data use policy</a></span></div><div class=\\"right-al\\"><span><a href=\\"/contacts\\">Make contact with us</a></span><span>Â© Shake it, Bulgaria 2019</span></div></footer>"`;
